frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts

```diff
@@ -108,6 +112,22 @@ export const propertyFilterLogic = kea<propertyFilterLogicType>({
 
     selectors: {
         filledFilters: [(s) => [s.filters], (filters) => filters.filter(isValidPropertyFilter)],
+        filters: [
+            (s) => [s.rawFilters, s.findPropertyDefinitionsByName],
+            (rawFilters, findPropertyDefinitionsByName) => {
+                return rawFilters.map((rawFilter) => {
+                    const processedFilter = { ...rawFilter }
+                    if (rawFilter.key && !rawFilter.property_type) {
+                        const propertyDefinition = findPropertyDefinitionsByName(rawFilter.key)
+                        if (propertyDefinition?.property_type) {
+                            processedFilter.property_type = propertyDefinition.property_type
+                            processedFilter.property_type_format = propertyDefinition.property_type_format
+                        }
+                    }
+                    return processedFilter
+                })
+            },
```

Comment:
Unfortunately this won't work. You will absolutely need to make an API call to get this information. That's because of two things: 

1) We have users with hundreds of thousands of property definitions, for whom they all gradually load and trickle in over many minutes after the app loads. 
2) We will remove that horrible logic soon enough (part of taxonomy), and this `propertyDefinitionsModel` will be empty or just contain the last page.